jx_sqlite/insert_table.py

```diff
@@ -290,28 +290,30 @@ def _flatten(data, uid, parent_id, order, full_path, nested_path, row=None, guid
                         )
                     for i, r in enumerate(v):
                         child_uid = self.next_uid()
-                        _flatten(r, child_uid, uid, i, cname, deeper_nested_path)
+                        _flatten(r, child_uid, uid, i, cname, deeper_nested_path, table=concat_field(self.sf.fact, cname))
                 elif value_type == "object":
                     row[c.es_column] = "."
-                    _flatten(v, uid, parent_id, order, cname, nested_path, row=row)
+                    _flatten(v, uid, parent_id, order, cname, nested_path, row=row, table=table)
                 elif c.type:
                     row[c.es_column] = v
-
+        
+        commands = []
         for doc in docs:
             _flatten(doc, self.next_uid(), 0, 0, full_path=path, nested_path=["."], guid=self.next_guid())
             if required_changes:
                 self.sf.change_schema(required_changes)
+                commands.append(required_changes[-1])
             required_changes = []
 
-        return doc_collection
+        return (doc_collection, commands)
 
     def next_uid(self):
         try:
             return self._next_uid
         finally:
             self._next_uid += 1
 
-    def _insert(self, collection):
+    def _insert(self, collection, commands):
```

Comment:
Rather than commands, use the `required_changes` list and add some `{"move": params}` instructions.  The instructions (however you may define them) can then be sent to the schema:

    self.sf.change_schema(required_changes)